### ISSUE TITLE
Make inheritdoc the default comment specification for interface and override methods

### ISSUE THREAD

**Issue by jjxtra** [dotnet/sdk#20858]
A developer has to add `inheritdoc` to all interface and override methods. Most of the time, you just need the comments from the base interface/virtual method comments.

Please make `<inheritdoc />` the default comment for interface and override methods.

Example:

```
public interface IStuff
{
    /// <summary>Do some stuff</summary>
    void DoStuff();
}

public class Stuff : IStuff
{
    public void DoStuff(); // yay! I get <inheritdoc/> on this method for free, but could also provide my own comments if desired.
}
```

Thanks for listening :)

**Comment by john-rubin** [comment]
This would be a great time-saver.

**Comment by sharwell** [comment]
This is already how `<inheritdoc/>` works. If a member does not have any comment, it is treated as having `<inheritdoc/>`. The implementation for automatic inheritdoc is here:
https://github.com/dotnet/roslyn/blob/80ea1700ed96e6441485e5a27319edd2b998f133/src/Workspaces/Core/Portable/Shared/Extensions/ISymbolExtensions.cs#L257-L260

**Comment by jjxtra** [comment]
I get compile warnings whenever I don't have a comment if I've setup comments to be warnings for this case, is there a way to have it detect that it's using this inheritance?

**Comment by sharwell** [comment]
The C# compiler doesn't understand `<inheritdoc/>`. The recommendation I always give is to disable the three compiler warnings related to missing comment elements. Here's an example:

https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/a559b3e58af09387aadc1372a504f9208bce4909/StyleCop.Analyzers/Directory.Build.props#L28-L39

**Comment by jjxtra** [comment]
To be clear, if I follow those guidelines, the warnings will still appear properly for any methods that do not implement an interface or override a virtual method?

**Comment by sharwell** [comment]
No, the warning will never appear if it is disabled. If you want to keep this warning, you could implement a custom DiagnosticSuppressor to remove the compiler warnings only from specific locations. Here's an example of a DiagnosticSuppressor I implemented for a different diagnostic:
https://github.com/dotnet/roslyn-analyzers/blob/main/src/Roslyn.Diagnostics.Analyzers/Core/RelaxTestNamingSuppressor.cs
